### PR TITLE
pypy: fixes for Linuxbrew

### PR DESCRIPTION
**Pull request by maxim-belkin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/pypy.rb [hunk 1]
```diff
@@ -65,6 +67,11 @@ def install
       python = buildpath/"bootstrap/bin/pypy"
     end
 
+    inreplace "lib_pypy/_tkinter/tklib_build.py" do |s|
+      s.gsub! "/usr/include/tcl", "#{Formula["tcl-tk"].opt_include}"
+      s.gsub! "'tcl' + _ver, 'tk' + _ver", "'tcl8.6', 'tk8.6'"
+    end
```
Comment (iMichka) [inline Formula/pypy.rb]: unless OS.mac?

**Comment by maxim-belkin** [comment]
I think the build on Circle times out rather than runs out of memory.

**Comment by iMichka** [comment]
Not sure. Normally it times out after 5 hours.

Your best bet is to connect to the running container with ssh and run a build on circle-ci to see what happens.

**Comment by iMichka** [comment]
The build which I ran with ssh worked. But the circle.ci jobs tells me it has failed with `Too long with no output (exceeded 1h0m0s)`. Maybe it is just that timeout which is the problem. We can try to set it to 3 hours in the circle.ci config and see what happens.

**Comment by maxim-belkin** [comment]
@iMichka, your build with `ssh` failed with a weird error:
```
=> Downloading https://bitbucket.org/pypy/pypy/downloads/pypy2-v6.0.0-linux64.tar.bz2
Already downloaded: /home/linuxbrew/.cache/Homebrew/pypy--bootstrap-6.0.0.tar.bz2
==> Verifying pypy--bootstrap-6.0.0.tar.bz2 checksum
tar xjf /home/linuxbrew/.cache/Homebrew/pypy--bootstrap-6.0.0.tar.bz2
==> FAILED
```
after seeing which I enabled "true" verbosity (unset `HOMEBREW_VERBOSE_USING_DOTS`) and set parallelism to "1", which produced the error message you quoted above. In my tests I never had an hour without output, so something is not right.

File: Formula/pypy.rb [hunk 2]
```diff
@@ -65,10 +67,15 @@ def install
...
+    inreplace "lib_pypy/_tkinter/tklib_build.py" do |s|
+      s.gsub! "/usr/include/tcl", Formula["tcl-tk"].opt_include.to_s
+      s.gsub! "'tcl' + _ver, 'tk' + _ver", "'tcl8.6', 'tk8.6'"
+    end unless OS.mac?
+
     cd "pypy/goal" do
       system python, buildpath/"rpython/bin/rpython",
              "-Ojit", "--shared", "--cc", ENV.cc, "--verbose",
-             "--make-jobs", ENV.make_jobs, "targetpypystandalone.py"
+             "--make-jobs", ENV["CIRCLECI"].nil? ? ENV.make_jobs : "4", "targetpypystandalone.py"
```
Comment (sjackman) [inline Formula/pypy.rb]: You can use instead…
```ruby
    # Reduce memory usage for CircleCI.
    ENV["MAKEFLAGS"] = "-j4" if ENV["CIRCLECI"]
```
which sets `ENV.make_jobs`

**Comment by sjackman** [comment]
Thanks for these fixes, Maxim!

File: Formula/pypy.rb (hunk 2 above)
Comment (maxim-belkin) [reply Formula/pypy.rb]: Let me try this instead. I thought `MAKEFLAGS` affect flags for `make` commands

Comment (sjackman) [reply Formula/pypy.rb]: `ENV.make_jobs` is derived from `ENV["MAKEFLAGS"]`
```ruby
  def make_jobs
    self["MAKEFLAGS"] =~ /-\w*j(\d+)/
    [Regexp.last_match(1).to_i, 1].max
  end
```
https://github.com/Linuxbrew/brew/blob/b9917814b14a54406a1e8588f4157e97cdc88c32/Library/Homebrew/extend/ENV/super.rb#L286

Comment (maxim-belkin) [reply Formula/pypy.rb]: So, `ENV.make_jobs` is an `int`

**Comment by maxim-belkin** [comment]
MAKEFLAGS do not help. I suggest we merge the actual fixes to the formula (without the CircleCI/verbosity stuff) and either build a bottle manually, or don't build it at all (what, perhaps, makes some sense)

**Comment by iMichka** [comment]
I'll merge this as-is and build the bottle locally.

I know that circle has now the option to change the hardware configuration:
https://circleci.com/docs/2.0/configuration-reference/#resource_class
It is only accessible to paid accounts. It would maybe be interesting to know how much they would charge us for this feature. We do not need so many parallel containers.

Because we have now more and more bottles I need to build myself, and it is quite a tedious work ...

**Comment by maxim-belkin** [comment]
🎉

**Comment by iMichka** [comment]
I'll build this #8206 locally tomorrow.

**Pull request closed without merging**